File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.cpp
class Solution {
public:
    int nearestValidPoint(int x, int y, vector<vector<int>>& points) {
        int min = INT_MAX;
        for(int i = 0; i < points.size(); i++) {
            if( (x == points[i][0]) || (y == points[i][1]) ) {
                int d = abs(points[i][0] - x) + abs(points[i][1] - y);
                if(d < min) min = d; 
            }
        }

        for(int i = 0; i < points.size(); i++) {
            if( (x == points[i][0]) || (y == points[i][1]) ) {
                int d1 = abs(points[i][0] - x) + abs(points[i][1] - y); 
                if(d1 == min) return i;
            }
        }
        return -1;
    }
};

File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        int ans=-1;
        int m=Integer.MAX_VALUE;
        for(int i=0;i<points.length;i++){
            if(points[i][0]==x || points[i][1]==y){
                int c=Math.min(points[i][0] , points[i][1]);
                if(x==points[i][0]){
                    if(m>Math.abs(points[i][1]-y)) {m=Math.abs(points[i][1]-y);
                    ans=i;}
                }
                else{
                    if(m>Math.abs(points[i][0]-x)) {m=Math.abs(points[i][0]-x);
                    ans=i;}
                }

            }
        }
        return ans;
    }
}